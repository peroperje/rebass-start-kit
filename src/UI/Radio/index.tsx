import * as forms from "@rebass/forms";

export type RadioProps = forms.RadioProps;
const Radio = forms.Radio;

export default Radio;
