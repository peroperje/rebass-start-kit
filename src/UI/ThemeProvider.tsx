import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
