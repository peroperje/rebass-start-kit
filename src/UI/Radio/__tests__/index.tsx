import React from 'react';
import render from 'react-test-renderer';

import Radio from '../index';
import ThemeProvider from '../../ThemeProvider';

describe('Radio', () => {
    it('Default snapshot', () => {
        const tree = render
            .create(
                <ThemeProvider>
                    <Radio>Test</Radio>
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
