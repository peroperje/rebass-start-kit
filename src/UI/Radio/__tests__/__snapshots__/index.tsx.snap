// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Default snapshot 1`] = `
.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  width: 24px;
  height: 24px;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 9999px;
  color: #dddddf;
}

input:checked ~ .emotion-1 {
  display: block;
}

input:checked ~ .emotion-1 {
  color: #07c;
}

input:focus ~ .emotion-1 {
  background-color: hsla(205,100%,40%,0.125);
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 24px;
  height: 24px;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 9999px;
  color: #dddddf;
}

input:checked ~ .emotion-2 {
  display: none;
}

input:checked ~ .emotion-2 {
  color: #07c;
}

input:focus ~ .emotion-2 {
  background-color: hsla(205,100%,40%,0.125);
}

<div
  className="emotion-3"
>
  <input
    className="emotion-0"
    type="radio"
  >
    Test
  </input>
  <svg
    aria-hidden="true"
    className="emotion-1"
    fill="currentcolor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="emotion-2"
    fill="currentcolor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
</div>
`;
